### PR TITLE
Fix comments to SBL core MemAllocLib to reflect code behavior

### PR DESCRIPTION
**Pull request by rprangar**

### CODE REVIEW

File: BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c [hunk 1]
```diff
@@ -54,8 +54,9 @@ InternalUpdateMemPoolBottom (
   Allocates a buffer of type EfiBootServicesData.
 
   Allocates the number bytes specified by AllocationSize of type EfiBootServicesData and returns a
-  pointer to the allocated buffer.  If AllocationSize is 0, then a valid buffer of 0 size is
-  returned.  If there is not enough memory remaining to satisfy the request, then NULL is returned.
+  pointer to the allocated buffer.  If AllocationSize is 0, LdrGlobal->MemPoolCurrTop is returned.
+  If there is not enough memory remaining to satisfy the request, InternalUpdateMemPoolTop ASSERTS 
+  and the function does not return.
```
Comment (gdong1) [inline BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c]: yes. this is the current code behavior to ASSERT even in release build. 
Do we have plan to support to build ASSERT out in release mode? If yes, the function will return.

Comment (mauricema) [reply BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c]: We have no plan to remove the ASSERT in release build for SBL.

Comment (rprangar) [reply BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c]: I think this is a good feedback. I can change the assert to deadloop.

Comment (gdong1) [reply BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c]: OK. then it is fine.

**Pull request merged**